File: cpp-tiny/lib/Models/EmojiBase.cpp


#include "EmojiBase.h"

using namespace Tiny;

EmojiBase::EmojiBase()
{
	emoji_code = std::string();
	emoji_name = std::string();
	reaction_type = std::string();
}

EmojiBase::EmojiBase(std::string jsonString)
{
	this->fromJson(jsonString);
}

EmojiBase::~EmojiBase()
{

}

void
EmojiBase::fromJson(std::string jsonObj)
{   
    bourne::json object = bourne::json::parse(jsonObj);

    const char *emoji_codeKey = "emoji_code";
    
    if(object.has_key(emoji_codeKey)) 
    {
        bourne::json value = object[emoji_codeKey];


        
        jsonToValue(&emoji_code, value, "std::string");


    }

    const char *emoji_nameKey = "emoji_name";
    
    if(object.has_key(emoji_nameKey)) 
    {
        bourne::json value = object[emoji_nameKey];


        
        jsonToValue(&emoji_name, value, "std::string");


    }

    const char *reaction_typeKey = "reaction_type";
    
    if(object.has_key(reaction_typeKey)) 
    {
        bourne::json value = object[reaction_typeKey];


        
        jsonToValue(&reaction_type, value, "std::string");


    }


}

bourne::json
EmojiBase::toJson()
{
    bourne::json object = bourne::json::object();

    



    object["emoji_code"] = getEmojiCode();


    



    object["emoji_name"] = getEmojiName();


    



    object["reaction_type"] = getReactionType();



    return object;

}

std::string
EmojiBase::getEmojiCode()
{
	return emoji_code;
}

void
EmojiBase::setEmojiCode(std::string  emoji_code)
{
	this->emoji_code = emoji_code;
}

std::string
EmojiBase::getEmojiName()
{
	return emoji_name;
}

void
EmojiBase::setEmojiName(std::string  emoji_name)
{
	this->emoji_name = emoji_name;
}

std::string
EmojiBase::getReactionType()
{
	return reaction_type;
}

void
EmojiBase::setReactionType(std::string  reaction_type)
{
	this->reaction_type = reaction_type;
}

File: java-vertx/src/main/java/org/openapitools/server/api/model/EmojiBase.java
package org.openapitools.server.api.model;

import java.util.Objects;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;

@JsonInclude(JsonInclude.Include.NON_NULL)
public class EmojiBase   {
  
  private String emojiCode;
  private String emojiName;
  private String reactionType;

  public EmojiBase () {

  }

  public EmojiBase (String emojiCode, String emojiName, String reactionType) {
    this.emojiCode = emojiCode;
    this.emojiName = emojiName;
    this.reactionType = reactionType;
  }

    
  @JsonProperty("emoji_code")
  public String getEmojiCode() {
    return emojiCode;
  }
  public void setEmojiCode(String emojiCode) {
    this.emojiCode = emojiCode;
  }

    
  @JsonProperty("emoji_name")
  public String getEmojiName() {
    return emojiName;
  }
  public void setEmojiName(String emojiName) {
    this.emojiName = emojiName;
  }

    
  @JsonProperty("reaction_type")
  public String getReactionType() {
    return reactionType;
  }
  public void setReactionType(String reactionType) {
    this.reactionType = reactionType;
  }


  @Override
  public boolean equals(Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    EmojiBase emojiBase = (EmojiBase) o;
    return Objects.equals(emojiCode, emojiBase.emojiCode) &&
        Objects.equals(emojiName, emojiBase.emojiName) &&
        Objects.equals(reactionType, emojiBase.reactionType);
  }

  @Override
  public int hashCode() {
    return Objects.hash(emojiCode, emojiName, reactionType);
  }

  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class EmojiBase {\n");
    
    sb.append("    emojiCode: ").append(toIndentedString(emojiCode)).append("\n");
    sb.append("    emojiName: ").append(toIndentedString(emojiName)).append("\n");
    sb.append("    reactionType: ").append(toIndentedString(reactionType)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }
}
